src/category_theory/limits/presheaf.lean

```diff
@@ -312,10 +312,27 @@ nat_iso_of_nat_iso_on_representables _ _ (hL ≪≫ (is_extension_along_yoneda _
 If `L` preserves colimits and `ℰ` has them, then it is a left adjoint. Note this is a (partial)
 converse to `left_adjoint_preserves_colimits`.
 -/
-def is_left_adjoint_of_preserves_colimits (L : (Cᵒᵖ ⥤ Type u₁) ⥤ ℰ) [preserves_colimits L] :
+def is_left_adjoint_of_preserves_colimits (L : (C ⥤ Type u₁) ⥤ ℰ) [preserves_colimits L] :
   is_left_adjoint L :=
-{ right := restricted_yoneda (yoneda ⋙ L),
-  adj := (yoneda_adjunction _).of_nat_iso_left
-              (unique_extension_along_yoneda _ L (iso.refl _)).symm }
+begin
+  let L' : (Cᵒᵖᵒᵖ ⥤ Type u₁) ⥤ ℰ := ((op_op_equivalence C).congr_left).functor ⋙ L,
+  have : preserves_colimits L',
+  { refine ⟨λ J 𝒥₁, _⟩,
+    constructor,
+    refine λ K, _,
+    apply_instance },
+  have : is_left_adjoint L',
+  { refine ⟨restricted_yoneda (yoneda ⋙ L'), _⟩,
+    resetI,
+    apply (yoneda_adjunction _).of_nat_iso_left _,
+    { apply_instance },
+    apply (unique_extension_along_yoneda _ L' (iso.refl _)).symm },
+  have : is_left_adjoint (((op_op_equivalence C).congr_left).inverse ⋙ L'),
```

Comment:
(Now that I look at it again, putting these as `have` rather than `let` might be a bad idea if you want to unfold the definition ever)

Reply:
Right, unless we decide that we shouldn't be relying on the definition ever, and always use the "unique-up-to-isom" API.
I'll let you decide.